addons/web/static/src/legacy/js/views/kanban/kanban_renderer.js

```diff
@@ -497,23 +497,34 @@ var KanbanRenderer = BasicRenderer.extend({
         var cleanGroupByField = this._cleanGroupByField(groupByField);
         var groupByFieldAttrs = this.state.fields[cleanGroupByField];
         var groupByFieldInfo = this.state.fieldsInfo.kanban[cleanGroupByField];
-        // Deactivate the drag'n'drop if the groupedBy field:
-        // - is a date or datetime since we group by month or
-        // - is readonly (on the field attrs or in the view)
+        // Deactivate the drag'n'drop:
+        // - if the groupedBy field is readonly (on the field attrs or in the view)
+        // - allow_drag_drop_create is false
+        // - for date and datetime if :
+        //   - allow_drag_drop_create is undefined
```

Comment:
I'd properly default this option in KanbanView, where you process the attrs. You know the field type there, so you can apply the correct default and ensure you manipulate a boolean in the rest of the code, which will make it easier.